.circleci/verbatim-sources/commands.yml

```diff
@@ -84,3 +84,24 @@ commands:
       - brew_update
       - brew_install:
           formulae: libtool
+
+  maybe_rebase:
+    steps:
+      - run:
+          name: Rebase on top of REBASE_TARGET if exists
+          no_output_timeout: "10m"
+          command: |
+            set -ex
+            if [ -n "$REBASE_TARGET" ] && [ -n "$CIRCLE_PULL_REQUEST" ]; then
+              git config --global user.email "circleci.ossci@gmail.com"
+              git config --global user.name "CircleCI"
+              PR_NUM=$(basename $CIRCLE_PULL_REQUEST)
+              CIRCLE_PR_BASE_BRANCH=$(curl -s https://api.github.com/repos/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME/pulls/$PR_NUM | jq -r '.base.ref')
+              echo "Rebase on top of $CIRCLE_PR_BASE_BRANCH branch before building in environment $BUILD_ENVIRONMENT"
+              if [[ "${CIRCLE_PR_BASE_BRANCH}" == "master" ]]; then
+                git rebase "${REBASE_TARGET}"
+              else
+                git rebase "origin/${CIRCLE_PR_BASE_BRANCH}"
```

Comment:
You should just merge with master here, there's no reason to actually do a rebase. The rebase is more likely to fail as each individual commit in the PR must merge cleanly.

We already have some existing logic for merging with master:

```
          # TODO We may want to move the rebase logic to a separate step after checkout
          # Rebase to master only if in xenial_py3_6_gcc5_4 case
          if [[ "${CIRCLE_BRANCH}" != "master" && "${BUILD_ENVIRONMENT}" == *"gcc5"* ]]; then
```

is there any reason this couldn't have been reused in this case?

Reply:
Hmmm no particular reason, I just thought rebase would be cleaner and easier than a merge :P 
Yea I can revert it back to merging :D

Reply:
Like I commented before, you should merge with whatever the base branch is for the PR.

If a base branch isn't available then it's a no-op.

Reply:
@seemethere What do you mean by `base branch for the PR`? Do you mean `CIRCLE_PR_BASE_BRANCH`? It's indeed `origin/master` if you are merging into master, and it'll be `release/1.5` if you want to merge into `release/1.5`. This is totally fine for release branches right? 
Note this code block is also guarded by `CIRCLE_PULL_REQUEST` that it only runs on PRs not branches.